src/logger.ts

```diff
@@ -76,26 +62,22 @@ export class Logger {
    * @param prefix {string} - the prefix of the logger
    */
   static withPrefix(prefix: string): Logging {
-    const loggerStuff = Logger.loggerCache.get(prefix);
+    const cachedLogging = Logger.loggerCache.get(prefix);
 
-    if (loggerStuff) {
-      return loggerStuff;
+    if (cachedLogging) {
+      return cachedLogging;
     } else {
       const logger = new Logger(prefix);
 
-      const log: IntermediateLogging = logger.info.bind(logger);
-      log.info = logger.info;
-      log.warn = logger.warn;
-      log.error = logger.error;
-      log.debug = logger.debug;
-      log.log = logger.log;
-
-      log.prefix = logger.prefix;
-
+      const logging = Object.assign(logger.info.bind(logger), {
+        info: logger.info,
+        warn: logger.warn,
+        error: logger.error,
+        debug: logger.debug,
+        log: logger.log,
+        prefix: prefix,
+      });
```

Comment:
Object assign returns `typeof arg1 & typeof arg2`, giving proper typing (when using `prefix` instead of `logger.prefix` as `prefix` on `Logger` is optional but we know it is defined for this function). This avoids the dangerous `@ts-ginore` assignment below.

Reply:
Thanks for resolving this 🚀